optimum/habana/transformers/models/llama/modeling_llama.py

```diff
@@ -654,9 +657,11 @@ def pre_attn_forward(
                     past_value = torch.zeros(
                         key_states.shape, dtype=self.get_k_proj_weight_dtype(), device=key_states.device
                     )
+                    past_key.copy_(key_states)
+                    past_value.copy_(value_states)
                     # Return list instead of tuple
```

Comment:
Could you explain more on this, what this needed for and how it was working before without this before.

Reply:
this is needed when num_virtual_token == max_seq_length in prefix-tuning. in previous logic, the past_value and past_key is a allocated as max_seq_length and self.k_cache.update will copy virtual token to past_value/key incorrectly in logic https://github.com/huggingface/optimum-habana/pull/1606/files#diff-30aeee6868dd1de34878aca0583f57bb5b0dd9a2a8511a80e9a6b2645f39ce6bL408. instead of using torch.cat, we do not have case that num_virtual_token==max_seq_length. so did not find the issue before.